cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/ZooKeeperEraser.java

```diff
@@ -0,0 +1,198 @@
+    private final NetworkPolicyOperator networkPolicyOperator;
+    private final PodDisruptionBudgetOperator podDisruptionBudgetOperator;
+
+    /**
+     * Constructs the ZooKeeper eraser
+     *
+     * @param reconciliation            Reconciliation marker
+     * @param config                    Cluster Operator Configuration
+     * @param supplier                  Supplier with Kubernetes Resource Operators
+     * @param kafkaAssembly             The Kafka custom resource
+     */
+    public ZooKeeperEraser(
+            Reconciliation reconciliation,
+            ClusterOperatorConfig config,
+            ResourceOperatorSupplier supplier,
+            Kafka kafkaAssembly
+    ) {
+        this.reconciliation = reconciliation;
+        // NOTE: we are going to use the eraser for deleting ZooKeeper so we are not interested in providing old storage and replicas parameters
+        this.zk = ZookeeperCluster.fromCrd(reconciliation, kafkaAssembly, config.versions(), null, 0, supplier.sharedEnvironmentProvider);
+        this.isNetworkPolicyGeneration = config.isNetworkPolicyGeneration();
+
+        this.strimziPodSetOperator = supplier.strimziPodSetOperator;
+        this.secretOperator = supplier.secretOperations;
+        this.serviceAccountOperator = supplier.serviceAccountOperations;
+        this.serviceOperator = supplier.serviceOperations;
+        this.pvcOperator = supplier.pvcOperations;
+        this.storageClassOperator = supplier.storageClassOperations;
+        this.configMapOperator = supplier.configMapOperations;
+        this.networkPolicyOperator = supplier.networkPolicyOperator;
+        this.podDisruptionBudgetOperator = supplier.podDisruptionBudgetOperator;
+    }
+
+    /**
+     * The main reconciliation method which triggers the whole reconciliation pipeline. This is the method which is
+     * expected to be called from the outside to trigger the reconciliation.
+     *
+     * @return              Future which completes when the reconciliation completes
+     */
+    public Future<Void> reconcile()    {
+        LOGGER.infoCr(reconciliation, "Deleting all the ZooKeeper related resources");
+        return jmxSecret()
+                .compose(i -> networkPolicy())
+                .compose(i -> serviceAccount())
+                .compose(i -> service())
+                .compose(i -> headlessService())
+                .compose(i -> certificateSecret())
+                .compose(i -> loggingAndMetricsConfigMap())
+                .compose(i -> podDisruptionBudget())
+                .compose(i -> podSet())
+                .compose(i -> deletePersistentClaims());
+    }
+
+    /**
+     * Deletes the secret with JMX credentials when JMX is enabled
+     *
+     * @return  Completes when the JMX secret is successfully deleted
+     */
+    protected Future<Void> jmxSecret() {
+        return secretOperator.deleteAsync(reconciliation, reconciliation.namespace(), zk.jmx().secretName(), true);
```

Comment:
`zk.jmx().secretName()` can be IMHO replaced by `KafkaResources.zookeeperJmxSecretName(<clusterName>)`

Reply:
Yeah, this is how the `JmxModel` get the secret name in the `ZooKeeperCluster.fromCrd` method. Going to replace.